### PR TITLE
100: Fixing jest testing that was not running as expected

### LINKED ISSUE #100: Create a Jest Sample For Unit Testing
Branched off of ID, will just be an open branch and not merged in. Will have sample test cases on how to test using Jest.

### PR DESCRIPTION
**Pull request by MitchWag01**
Closes #100 

# What was the issue
When running Jest, there was a problem where it was missing its Jestconfig file. This made testing not reliable and added
miss mathced testing where some things were javascript and some things were typescript

# What was done and why
1. Created a jestconfig file that sets Jest to typescript
2. Added extra compiler option to typescript config to allow imports to Jest.test files
3. Fixed tests that were using javascript

# How it was tested
To verify this solution i ran all sample test cases that were created -- these are being moved to the wiki for other developers to see how to write different style unit tests. The tests that were originally in the system were modified to the new standard and then ran, all tests passed successfully. 

Another scenario that was used was running the projects npm test, this showed that the project itself tests and runs as it originally should have.

### CODE REVIEW

File: my-webxr-app/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// jest.config.ts
+/** @type {import('jest').Config} */
+const config = {
+    preset: 'ts-jest',
+    testEnvironment: 'node',
+    // esModuleInterop: true,
+};
+
+export default config;
```
Comment (matthew-buglass) [inline my-webxr-app/jest.config.js:9]: save line of code by replacing line 3 with export const config

File: my-webxr-app/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// jest.config.ts
+/** @type {import('jest').Config} */
+const config = {
```
Comment (matthew-buglass) [inline my-webxr-app/jest.config.js:3]: have better var naming style jest config

File: my-webxr-app/jest.config.js
Comment (matthew-buglass) [inline my-webxr-app/jest.config.js:1]: Try to make it a typescript file

**Review by Joembonayo — APPROVED** [review]

File: my-webxr-app/jest.config.js (hunk 2 above)
Comment (MitchWag01) [reply my-webxr-app/jest.config.js]: This change has been completed

File: my-webxr-app/jest.config.js
Comment (MitchWag01) [reply my-webxr-app/jest.config.js:1]: Was unable to make this change as then we require babel for a single file to run

**Pull request merged**